### PR TITLE
doc: show-md on params/metrics diff

### PR DESCRIPTION
**Pull request by skshetry**
> You may disregard these recommendations if you used the **Edit on GitHub** button from dvc.org to improve a doc in place.

❗ Please read the guidelines in the [Contributing to the Documentation](https://dvc.org/doc/user-guide/contributing/docs) list if you make any substantial changes to the documentation or JS engine.

🐛 Please make sure to mention `Fix #issue` (if applicable) in the description of the PR. This causes GitHub to close it automatically when the PR is merged.

Please chose to [allow us](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork) to edit your branch when creating the PR.

Thank you for the contribution - we'll try to review it as soon as possible. 🙏

Implemented on https://github.com/iterative/dvc/pull/3757

### CODE REVIEW

**Comment by jorgeorpinel** [comment]
Thanks @skshetry. Can you please link to the core repo PR so we know whether or when it's merged? Bc I'm not seeing this behavior on dvc master branch yet.

File: content/docs/command-reference/metrics/diff.md [hunk 1]
```diff
@@ -9,7 +9,8 @@ commits in the <abbr>DVC repository</abbr>, or between a commit and the
 ```usage
 usage: dvc metrics diff [-h] [-q | -v]
                         [--targets [<path> [<path> ...]]]
-                        [-t <type>] [-x <path>] [-R] [--show-json]
+                        [-t <type>] [-x <path>] [-R]
+                        [--show-json] [--show-md]
                         [a_ref] [b_ref]
```
Comment (jorgeorpinel) [inline content/docs/command-reference/metrics/diff.md:14]: Is this the actual order in which it's printed? I don't see this output in the core repo master branch yet. Thanks

Comment (skshetry) [reply content/docs/command-reference/metrics/diff.md:14]: Yes, it wasn't merged then. Please take a look again.

File: content/docs/command-reference/metrics/diff.md [hunk 2]
```diff
@@ -105,7 +108,7 @@ had in the previous commit:
 
 ```
 $ dvc metrics diff
-    Path       Metric   Value   Change
-metrics.json   TP       531     4
-metrics.json   AUC      0.967   0.003
+Path          Metric    Value    Change
+metrics.json  TP        531      4
+metrics.json  AUC       0.967    0.003
```
Comment (skshetry) [inline content/docs/command-reference/metrics/diff.md:113]: The heading formats got affected by the change. Previously it was centered, it's left aligned now.

File: content/docs/command-reference/metrics/diff.md (hunk 1 above)
Comment (skshetry) [reply content/docs/command-reference/metrics/diff.md:14]: > Is this the actual order in which it's printed?

Yes, that's the order.

```sh
$ dvc metrics diff -h
usage: dvc metrics diff [-h] [-q | -v] [--targets [<paths> [<paths> ...]]] [-R] [--all] [--show-json] [--show-md]
                        [a_rev] [b_rev]

Show changes in metrics between commits
Documentation: <https://man.dvc.org/metrics/diff>

positional arguments:
  a_rev                 Old Git commit to compare (defaults to HEAD)
  b_rev                 New Git commit to compare (defaults to the current workspace)

optional arguments:
  -h, --help            show this help message and exit
  -q, --quiet           Be quiet.
  -v, --verbose         Be verbose.
  --targets [<paths> [<paths> ...]]
                        Metric files or directories (see -R) to show diff for. Shows diff for all metric files by default.
  -R, --recursive       If any target is a directory, recursively search and process metric files.
  --all                 Show unchanged metrics as well.
  --show-json           Show output in JSON format.
  --show-md             Show tabulated output in the Markdown format (GFM).
```

Note that we still need to readjust the docs for the `-t` and `-x` on the metrics. But, I have tried to keep the order for all `dvc {metrics,params} diff`.

**Comment by shcheklein** [comment]
@skshetry would be great to update bash/zsh scripts as well.

**Comment by skshetry** [comment]
@shcheklein, I have updated zsh/bash autocompletion scripts.

File: content/docs/command-reference/metrics/diff.md (hunk 1 above)
Comment (jorgeorpinel) [reply content/docs/command-reference/metrics/diff.md:14]: Looks like you're missing `[--all]` thouogh.

Comment (jorgeorpinel) [reply content/docs/command-reference/metrics/diff.md:14]: I'll take care of it.

**Pull request merged**